### PR TITLE
Visa Checkout and Masterpass

### PR DESCRIPTION
**Pull request by aengusbates**
Migrating Visa Checkout and Masterpass docs from support.na.bambora.com.

Quick and dirty migration that will need to revisited once we have the querystring API docs published. The speed is required so as not to block the timeline for retiring support.na.bambora.com.

Replacing: 

- https://support.na.bambora.com/bic/w/index.html#docs/masterpass-process-flow.htm%3FTocPath%3DDeveloper%2520Resources%7CMasterPass%2520Process%2520Flow%7C_____1
- https://support.na.bambora.com/bic/w/index.html#docs/visa-checkout-intro.htm%3FTocPath%3DDeveloper%2520Resources%7CVisa%2520Checkout%7C_____0

### CODE REVIEW

File: source/docs/guides/masterpass.md [hunk 1]
```diff
@@ -17,23 +17,85 @@ navigation:
...
-Masterpass leverages your existing environment because most websites already exist where Masterpass will 
-be used. You add Masterpass buttons to existing pages and implement business and event logic using programming 
-languages, tools, and techniques in the same way you currently do. This makes Masterpass flexible and imposes 
-only a few requirements.
+Masterpass simplifies and secures online payments by storing payment information.
+It removes the need for a customer to re-enter and share card information.
 
-Masterpass is only available for Merchants processing with either First Data or TD Online Mart. 
-Contact our support team (1-888-472-0811) for further information.
+Masterpass is a digital wallet. Customers load their digital wallet with their credit, debit and prepaid cards once. They can then retrieve their payment information at any checkout supporting the wallet.
```
Comment (davidlight77) [inline source/docs/guides/masterpass.md]: ...at any checkout that supports Masterpass.

File: source/docs/guides/masterpass.md [hunk 2]
```diff
@@ -17,23 +17,85 @@ navigation:
...
 
-If you have integrated and certified to Mastercard’s Masterpass API and are handling the redirect to the Masterpass Portal 
-independently of Bambora’s internal integration, you will need to pass the Masterpass Wallet ID with your Payment REST API 
+## Standard integration
+
+A Masterpass payment involves three steps for a merchant.
+
+1. Retrieve a redirect URL from our Payment API.
+2. Redirect the user to Masterpass and handle the return redirect from Masterpass.
+3. complete the payment on out Payment API.
```
Comment (davidlight77) [inline source/docs/guides/masterpass.md]: Complete should be capitalized and 'out' should be 'our'

File: source/docs/guides/masterpass.md [hunk 3]
```diff
@@ -17,23 +17,85 @@ navigation:
...
+
+### Cavets
```
Comment (davidlight77) [inline source/docs/guides/masterpass.md]: Caveats

File: source/docs/guides/masterpass.md [hunk 4]
```diff
@@ -17,23 +17,85 @@ navigation:
...
+##### Second response
+```shell
+trnApproved=1&trnId=10000011&messageId=1&messageText=Approved&trnOrderNumber=10000011&authCode=TEST&errorType=N&errorFields=&responseType=T&trnAmount=10%2E00&trnDate=8%2F27%2F2017+12%3A43%3A02+AM&avsProcessed=1&avsId=N&avsResult=0&avsAddrMatch=0&avsPostalMatch=0&avsMessage=Street+address+and+Postal%2FZIP+do+not+match%2E&cvdId=5&cardType=MC&trnType=P&paymentMethod=CC&ref1=&ref2=&ref3=&ref4=&ref5=&hashValue=1e48a7346bac9cb6da3dfc4e26d2cd9b2a3e5c6b
+```
+
+
+## Advanced integration
+
+If you have integrated and certified to Mastercard’s Masterpass API and are handling the redirect to the Masterpass Portal
+independently of Bambora’s internal integration, you will need to pass the Masterpass Wallet ID with your Payment REST API
```
Comment (davidlight77) [inline source/docs/guides/masterpass.md]: should this be one line?

File: source/docs/guides/visa_checkout.md [hunk 5]
```diff
@@ -17,23 +17,100 @@ navigation:
...
+
+## Standard integration
+
+A Visa Checkout payment can involve either one or two steps for a merchant.
+
+It will always begin with a user clicking on a Visa Checkout button. Visa Checkout's
+SDK handles the event and pops a lightbox where the user can authenticate and select a payment card.
+Visa checkout then calls a callback function on the merchant's webpage.
+
+The one step process involves simple listening for a callback from Visa's SDK and calling our Payment API to complete the payment.
```
Comment (davidlight77) [inline source/docs/guides/visa_checkout.md]: ...simply...

File: source/docs/guides/visa_checkout.md [hunk 6]
```diff
@@ -17,23 +17,100 @@ navigation:
...
+
+The two step process involves listening for a callback from Visa's SDK, calling
+our API to retrieve addresses, and then calling our Payment API to complete the payment.
+
+### Cavets
```
Comment (davidlight77) [inline source/docs/guides/visa_checkout.md]: Caveats

File: source/docs/guides/visa_checkout.md [hunk 7]
```diff
@@ -17,23 +17,100 @@ navigation:
...
+
+Bambora's Visa Checkout integration is not compatible with 3D Secure.
+
+### Testing
+
+Visa Checkout is not enabled by default on test accounts. Contact us at (support.northamerica@bambora.com)[mailto:support.northamerica@bambora.com]
+and we will enable it.
```
Comment (davidlight77) [inline source/docs/guides/visa_checkout.md]: newline not needed?

File: source/docs/guides/visa_checkout.md [hunk 8]
```diff
@@ -17,23 +17,100 @@ navigation:
...
+
+You can test visa checkout on sandbox-web.na.bambora.com.
```
Comment (davidlight77) [inline source/docs/guides/visa_checkout.md]: Should be a link

File: source/docs/guides/visa_checkout.md [hunk 9]
```diff
@@ -17,23 +17,100 @@ navigation:
...
+avsMessage=Street+address+and+Postal%2FZIP+do+not+match%2E&cvdId=5&
+ref1=&ref2=&ref3=&ref4=&ref5=&trnType=P&paymentMethod=CC&
+trnDate=8%2F26%2F2017+11%3A54%3A32+PM&httpStatusCode=200&errorCategory=1&
+cardType=VI&cardLastFour=1234&trnAmount=10.00&authorizingMerchantId=123456789
+```
+
+## Advanced integration
+
+If you have integrated and certified to Visa’s Visa Checkout API and are handling the redirect to the Visa Checkout Portal
+independently of Bambora’s internal integration, you will need to pass the Visa Checkout Call ID with your Payment REST API
```
Comment (davidlight77) [inline source/docs/guides/visa_checkout.md]: newline not needed?

File: source/docs/guides/visa_checkout.md (hunk 8 above)
Comment (aengusbates) [reply source/docs/guides/visa_checkout.md]: Hyperlinks redirect the browser. This string is the base URL of an API. It doesn't return a HTML page.

File: source/docs/guides/visa_checkout.md (hunk 9 above)
Comment (aengusbates) [reply source/docs/guides/visa_checkout.md]: New lines are not a issue in Markdown. They make the source more readable in text editors, but don't render an differently.

File: source/docs/guides/visa_checkout.md (hunk 8 above)
Comment (aengusbates) [reply source/docs/guides/visa_checkout.md]: Same comment. This is not a link. It doesn't return a HTML page.

Comment (davidlight77) [reply source/docs/guides/visa_checkout.md]: OK

File: source/docs/guides/masterpass.md [hunk 10]
```diff
@@ -34,9 +34,9 @@ A Masterpass payment involves three steps for a merchant.
 
 1. Retrieve a redirect URL from our Payment API.
 2. Redirect the user to Masterpass and handle the return redirect from Masterpass.
-3. complete the payment on out Payment API.
+3. Complete the payment on out Payment API.
```
Comment (davidlight77) [inline source/docs/guides/masterpass.md]: still says 'out'

File: source/docs/guides/visa_checkout.md (hunk 7 above)
Comment (aengusbates) [reply source/docs/guides/visa_checkout.md]: These new lines will not be rendered as new lines are note rendered in Markdown. You need two consecutive new line chars to break paragraphs. 

These new lines are purely for readability in text editors. Some text editors don't wrap lines by default. Let's leave this for the moment and see what text editors are affected and when.

**Pull request merged**